<!Doctype html>
<html>
    <head>
        <meta charset="UTF-8" />
        <title>Seqphase</title>
        <link rel="stylesheet" href="style.css"/>
    </head>
    <body>
        <div class="contentdiv">
            <h1>SeqPHASE</h1>
            <p id="cite">Flot (2010) <strong>SeqPHASE: a web tool for interconverting PHASE input/output files and FASTA sequence alignments </strong> <em>Molecular Ecology Ressources</em> <strong>10</strong> (1): 162-166 [<a href="http://dx.doi.org/10.1111/j.1755-0998.2009.02732.x">link</a>]</p><br>
            <div class="navigation">
                <a href="index.html">Back to startpage</a>
                <a href="faq.html">FAQ</a>
                <a href="examples.zip">Download examples</a>
                <a href="step2.html">Step 2</a>
                <a href="mailto: jflot@ulb.ac.be">E-Mail the author</a>
            </div>
            <h2>Step 1: generating PHASE input files from FASTA alignments</h2>
            <div class="tipSection">
                <p><strong>Important tips!!</strong>
                <ol>
                    <li>Unlike many other programs, PHASE treats very differently indels and missing data. Hence, it is important to code missing data as "N" or "?" in your FASTA alignments and not as "-", which should be reserved for "real" indels.</li>
                    <li>If you input more than one FASTA alignment (e.g. because of have already determined the haplotypes of some length-variant heterozygotes using <a href="http://jfflot.mnhn.fr/champuru/">Champuru</a>), you should align all your sequences together first before splitting them into the different FASTA files. If you align each of them separately and the resulting alignments do not match, PHASE may not infer the haplotypes correctly.</li>
                </ol>
            </div>
            <table>
                <tr class="option">
                    <td>Genotypes to be phased</td>
                    <td>&nbsp;&nbsp;&nbsp;</td>
                    <td><input id="alig1" type="file" autocomplete="off"></td>
                </tr>
                <tr class="description">
                    <td colspan="3">
                        <p>Alignment of sequences from homozygous individuals and from heterozygotes to be phased (FASTA, one sequence of IUPAC one-letter codes per individual)</p>
                    </td>
                </tr>
                <tr class="option">
                    <td>Fake haplotype pairs</td>
                    <td>&nbsp;&nbsp;&nbsp;</td>
                    <td><input id="alig2" type="file" autocomplete="off"></td>
                </tr>
                <tr class="description">
                    <td colspan="3">
                        <p>Alignment of fake haplotype pairs from heterozygotes to be phased (FASTA, two sequences per individual). In this alignment, sequences for a given individual should have the same name except for the last character (for instance: "sample1a" for the first haplotype of sample 1, "sample1b" for the other).</p>
                        <p>Fake haplotype pairs are an alternate way of entering genotypes to be phased that is particularly useful when dealing with length-variant heterozygotes (since there is no IUPAC sign for "A or indel", "C or indel", etc.), but I rather recommend solving such cases first using <a href="http://jfflot.mnhn.fr/champuru/">Champuru</a> then inputting the result as known haplotype pairs in the 3rd SeqPHASE input field (below).</p>
                    </td>
                </tr>
                <tr class="option">
                    <td>Known haplotype pairs</td>
                    <td>&nbsp;&nbsp;&nbsp;</td>
                    <td><input id="alig3" type="file" autocomplete="off"></td>
                </tr>
                <tr class="description">
                    <td colspan="3">
                        <p>Alignment of known haplotype pairs, if any (FASTA, two sequences per individual).<br>In this alignment, sequences for a given individual should have the same name except for the last character (for instance: "sample1a" for the first haplotype of sample 1, "sample1b" for the other).</p>
                    </td>
                </tr>
            </table>
            <span class="middle">
                <button id="submitJob1Button">Run SeqPHASE</button>
                <button id="resetJob1Button">Reset Form</button>
            </span>
            <div id="align1FileErrorsArea" style="display: none">
                <br>
                The following <b>errors</b> occurred while processing the <i>one sequence per individual input file</i>:<br>
                <ol id="align1FileErrorsList">
                </ol>
            </div>
            <div id="align1FileWarningsArea" style="display: none">
                The following <b>warnings</b> occurred while processing the <i>one sequence per individual input file</i>:<br>
                <ol id="align1FileWarningsList">
                </ol>
            </div>
            <div id="align2FileErrorsArea" style="display: none">
                <br>
                The following <b>errors</b> occurred while processing the <i>two sequence per individual input file</i>:<br>
                <ol id="align2FileErrorsList">
                </ol>
            </div>
            <div id="align2FileWarningsArea" style="display: none">
                The following <b>warnings</b> occurred while processing the <i>two sequence per individual input file</i>:<br>
                <ol id="align2FileWarningsList">
                </ol>
            </div>
            <div id="align3FileErrorsArea" style="display: none">
                <br>
                The following <b>errors</b> occurred while processing the <i>file of known haplotype pairs</i>:<br>
                <ol id="align3FileErrorsList">
                </ol>
            </div>
            <div id="align3FileWarningsArea" style="display: none">
                The following <b>warnings</b> occurred while processing the <i>file of known haplotype pairs</i>:<br>
                <ol id="align3FileWarningsList">
                </ol>
            </div>
            <div id="generalErrorsArea" style="display: none">
                <br>
                The following <b>errors</b> occurred:
                <ol id="generalErrorsList">
                </ol>
            </div>
            <div id="generalWarningsArea" style="display: none">
                The following <b>warnings</b> occurred:
                <ol id="generalWarningsList">
                </ol>
            </div>
            <div id="notesArea" style="display: none">
            </div>
            <div id="outFileResult" style="display: none">
                <p>Since PHASE only accepts numbers and not letters for nucleotides at multistate characters,
                ? and N (missing information) have been replaced with ? or -1 (depending on whether the
                position displays two or more than two different nucleotides), - with 0, A with 1, C with 2,
                G with 3 and T with 4 in <a id="downloadInpFileLink" download="seqphase.inp">seqphase.inp</a>
                (the main PHASE input file).</p>
            </div>
            <div id="knownFileResult" style="display: none">
                <p>A <a id="downloadKnownFileLink" download="seqphase.known">known phase file</a>
                has also been generated to tell PHASE which phases are already known and which ones are
                to infer.</p>
            </div>
            <div id="constFileResult" style="display: none">
                <p>In order to reduce PHASE running time, constant positions have been removed from the
                dataset. These constant positions have been stored in
                <a id="downloadConstFileLink" download="seqphase.const">seqphase.const</a>
                interspersed with periods (.) representing variable positions.</p>
            <div id="suggestedCommandArea" style="display: none">
                <br>
                Suggested command syntax (PHASE v2.1):
                <span style="float:right;">[<a id="copySuggestedCommand" style="color: blue; text-decoration: underline;">Click here to copy suggested command to clipboard</a>]</span><br>
                <span class="middle">
                    <input id="suggestedCommandField" type="text" readonly="readonly" style="width: 720px; text-align: center" />
                </span>
                (the result of the analysis will be stored in a series of files with names starting with "seqphase.out").
            </div>
            <div id="proceedArea" style="display: none">
                <p>Once PHASE has been run, <a id="finalLink">please proceed here</a> to convert the PHASE
                output into a FASTA alignment (only the file seqphase.out is needed for this final step).
                (The above link will automatically pass the content of the .const file - if any -
                to the next step.)</p>
            </div>
            <br>
            <br>
            <div class="navigation">
                <a href="index.html">Back to startpage</a>
                <a href="faq.html">FAQ</a>
                <a href="examples.zip">Download examples</a>
                <a href="step2.html">Step 2</a>
                <a href="mailto: jflot@ulb.ac.be">E-Mail the author</a>
            </div>
        </div>
        <script src="seqphase1.js" type="text/javascript"></script>
        <script src="step1.js" type="text/javascript"></script>
    </body>
</html>
